vms/platformvm/utxo/camino_locked.go

```diff
@@ -707,6 +935,301 @@ func (h *handler) VerifyLockUTXOs(
+		if !ok {
+			consumedOwnerAmounts = make(map[ids.ID]uint64)
+			consumed[ownerID] = consumedOwnerAmounts
+		}
+
+		newAmount, err := math.Add64(consumedOwnerAmounts[lockIDs.BondTxID], consumedAmount)
+		if err != nil {
+			return nil, err
+		}
+		consumedOwnerAmounts[lockIDs.BondTxID] = newAmount
+	}
+
+	// iterating over outs, checking produced amounts with consumed map
+	// filling deposit produced amounts
+
+	for _, output := range outs {
+		out := output.Out
+		lockIDs := &locked.IDsEmpty
+		if lockedOut, ok := out.(*locked.Out); ok {
+			lockIDs = &lockedOut.IDs
+			out = lockedOut.TransferableOut
+		}
+
+		ownerID, err := GetOwnerID(out)
+		if err != nil {
+			return nil, err
+		}
+
+		producedAmount := out.Amount()
+		consumedAmount := uint64(0)
+		consumedOwnerAmounts, ok := consumed[ownerID]
+		if ok {
+			consumedAmount = consumedOwnerAmounts[lockIDs.BondTxID]
+		}
+
+		if consumedAmount < producedAmount {
+			return nil, fmt.Errorf(
+				"address %s produces %d and consumes %d for lockIDs %+v with lock '%s': %w",
+				ownerID,
+				producedAmount,
+				consumedAmount,
+				lockIDs.BondTxID,
+				locked.StateDeposited,
+				errWrongProducedAmount,
+			)
+		}
+
+		consumedOwnerAmounts[lockIDs.BondTxID] = consumedAmount - producedAmount
+
+		if depUnlock, ok := depositUnlocks[lockIDs.DepositTxID]; ok {
+			newAmount, err := math.Add64(depUnlock.producedAmount, producedAmount)
+			if err != nil {
+				return nil, err
+			}
+			depUnlock.producedAmount = newAmount
+		}
+	}
+
+	// this map will list how much tokens was unlocked from each deposit
+	unlockedAmount := make(map[ids.ID]uint64) // depositTxID -> amount
+
+	for depositTxID, depUnlock := range depositUnlocks {
+		if depUnlock.consumedAmount < depUnlock.producedAmount {
+			return nil, errWrongProducedAmount
+		}
+
+		unlockedDepositAmount := depUnlock.consumedAmount - depUnlock.producedAmount
+
+		// if we don't need keys, than deposit is expired and must be fully unlocked
+		// that means that tx must fully consume remaining deposited tokens and
+		// produce them as unlocked
+		if !depUnlock.needKeys &&
+			(unlockedDepositAmount != depUnlock.unlockableAmount ||
+				depUnlock.consumedAmount != depUnlock.unlockableAmount) {
+			return nil, fmt.Errorf("expired deposit (%s) unlockable amount (%d) isn't equal to consumed (%d) and produced (%d) amount: %w",
+				depositTxID,
+				depUnlock.unlockableAmount,
+				depUnlock.consumedAmount,
+				depUnlock.producedAmount,
+				errNotConsumedDeposit)
+		}
+
+		// checking that we unlocked no more, than was available for unlock
+		if unlockedDepositAmount > depUnlock.unlockableAmount {
+			return nil, fmt.Errorf("unlockedDepositAmount %d > %d unlockableAmount: %w",
+				unlockedDepositAmount,
+				depUnlock.unlockableAmount,
+				errUnlockedMoreThanAvailable)
+		}
+
+		unlockedAmount[depositTxID] = unlockedDepositAmount
+	}
+
+	// checking that we burned required amount
+
+	amountToBurn := burnedAmount
```

Comment:
amountToBurn should be actually renamed to "burnedAmount". I would suggest initializing it with 0 and adding up in the following for loop.

Reply:
well, thats arguable, cause we'r actually verifying already sort of happend situation, where some amount assumed to be burned, thats why arg is burnedAmount, not amount to burn - we assume its already burned with this ins/outs composition